src/App.css

```diff
@@ -676,6 +676,39 @@ a.text-link {
   word-break: break-word; /* breaks urls; backwards-compatible with regular text unlike break-all */
 }
 
+/* Make sure the parent containers of URLs fill the full width to prevent ellipsizing single-line URLs when not actively editing. */
+.thought:has(.url:not(.active)),
+.thought-annotation:has(.url:not(.active)),
+.editable-annotation:has(.url:not(.active)) {
+  width: 100%;
+}
+
+/* Ellipsize urls when not actively editing. */
+.editable.url:not(.active),
+.editable-annotation-text.url:not(.active) {
+  max-width: calc(100% - 2em);
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+/* Allow the URL icon to not affect box size when not actively editing. */
+.editable-annotation-text.url:not(.active) {
+  display: inline-block;
+}
+
+/* When actively editing, account for URL icon space. */
+.thought:has(.url.active),
+.thought-annotation:has(.url.active) {
+  width: calc(100% - 2.34em);
+}
+
+/* Adjust multiline icon alignment */
+.editable-annotation.multiline svg {
+  /* This rule requires !important to override the applied margin in the inline styles. */
+  margin-top: -0.08em !important;
```

Comment:
Please find another way to accomplish this besides `!important`. It makes the css harder to maintain if it needs to be overridden in the future.